File: WebServiceOfficePrivate/src/java/bonclub/office_private/web_service/IAccountSuperBon.java
package bonclub.office_private.web_service;

import java.util.List;

import bonclub.office_private.web_service.common_objects.GiftInformation;
import bonclub.office_private.web_service.common_objects.GiftOrderResult;
import bonclub.office_private.web_service.common_objects.Purse;
import bonclub.office_private.web_service.common_objects.PurseHistory;

import javax.jws.WebMethod; 
import javax.jws.WebParam; 
import javax.jws.WebService; 

@WebService
public interface IAccountSuperBon{
    @WebMethod
    public List<Purse> getBalance(
        @WebParam(name="id_club_card_purse") Integer id_club_card_purse);
    @WebMethod
    public List<PurseHistory> getPurseHistory(
        @WebParam(name="id_club_card_purse") Integer id_club_card_purse);

    @WebMethod
    public List<GiftInformation> getGiftInformation(
        @WebParam(name="id_club_card_purse") Integer id_club_card_purse);
    
    @WebMethod
    public GiftOrderResult sendOrderOfGift(
    	@WebParam(name="p_id_nat_prs") Integer p_id_nat_prs, 
    	@WebParam(name="p_id_club_card_purse") String p_id_club_card_purse, 
    	@WebParam(name="p_id_club_event_gift") String p_id_club_event_gift);
    
}

File: WebServiceOfficePrivate/src/php/IAccountSuperBon.php
<?php
class iaccountsuperbon{
    private $proxy;

    public function __construct($pUrl){
        $this->proxy = new SoapClient($pUrl);
    }

    public function getBalance($id_club_card_purse){
        $userData->id_club_card_purse=new SoapVar($id_club_card_purse, XSD_INT);
        return $this->proxy->__soapCall("getBalance", array($userData) );
    }
    public function getPurseHistory($id_club_card_purse){
        $userData->id_club_card_purse=new SoapVar($id_club_card_purse, XSD_INT);
        return $this->proxy->__soapCall("getPurseHistory", array($userData) );
    }

    public function getGiftInformation($id_club_card_purse){
        $userData->id_club_card_purse=new SoapVar($id_club_card_purse, XSD_INT);
        return $this->proxy->__soapCall("getGiftInformation", array($userData) );
    }

    public function sendOrderOfGift($p_id_nat_prs, $p_id_club_card_purse, $p_id_club_event_gift){
        $userData->p_id_nat_prs=new SoapVar($p_id_nat_prs, XSD_INT);
        $userData->p_id_club_card_purse=new SoapVar(urlencode($p_id_club_card_purse), XSD_STRING);
        $userData->p_id_club_event_gift=new SoapVar(urlencode($p_id_club_event_gift), XSD_STRING);
        return $this->proxy->__soapCall("sendOrderOfGift", array($userData) );
    }

}
?>